src/Controllers/WateruseController.ts

```diff
@@ -143,6 +163,21 @@ module StreamStats.Controllers {
         public Print(): void {
             window.print();
         }
+
+        private setIndexes(): any {
+            var Q10 = 0;
+            if(this.NSSService.selectedStatisticsGroupList) {
+                if(this.NSSService.selectedStatisticsGroupList[0]) {
+                    if(this.NSSService.selectedStatisticsGroupList[0].regressionRegions[0]) {
+                        if(this.NSSService.selectedStatisticsGroupList[0].regressionRegions[0].results[0] && this.AnnualTotalReturns != null && this.AnnualTotalWithdrawals != null && this.permits_gw.aveWithdrawal != null && this.permits_sw.aveWithdrawal != null) {
+                            Q10 = +this.NSSService.selectedStatisticsGroupList[0].regressionRegions[0].results[0].value * 0.646316889697;
+                            this._indexWOReg = (+this.AnnualTotalWithdrawals - +this.AnnualTotalReturns) / Q10;
+                            this._indexWReg = (+this.permits_gw.aveWithdrawal + +this.permits_sw.aveWithdrawal + (+this.AnnualTotalWithdrawals - +this.AnnualTotalReturns)) / Q10;
+                        }
+                    }

```

Comment:
Hm, if we can I'd like to clean this up. Have you tried putting these on one line, instead of multiple if statements?

Reply:
This was actually a remnant of a previous issue I was having looking into the wrong service - one of those copy/paste and forget to replace the value errors on my part. The properties kept coming up undefined and in the past I've had to do weird checks like this, breaking down each element and checking to see if it exists but now that I think about it that's only happened to me with the new angular. Anyway, just explaining the madness behind this so you don't think I'm crazy! ;-) I was able to shorten it considerably.